tests/component/test.headers.js

```diff
@@ -19,25 +22,43 @@ describe('test.headers.js', function () {
     server.listen(PORT, done);
   });
 
-  after(function (done) {
-    server.close(done);
+  after(function () {
+    return server.close();
   });
 
-
-  it('Test headers are sent correctly', function (done) {
+  it('Test headers are sent correctly', function () {
     var opts = {ajax: {headers: {foo: 'bar'}}};
-    new PouchDB('http://127.0.0.1:' + PORT, opts, function() {
-      assert.equal(headers.foo, 'bar');
-      done();
+    return new PouchDB('http://127.0.0.1:' + PORT, opts)
+    .then(function() {
+        should.equal(headers.foo, 'bar');
     });
   });
 
-  it('Test auth params are sent correctly', function (done) {
+  it('Test auth params are sent correctly', function () {
     var opts = {auth: {username: 'foo', password: 'bar'}};
-    new PouchDB('http://127.0.0.1:' + PORT, opts, function() {
-      assert.equal(typeof headers.authorization, 'string');
-      done();
+    return new PouchDB('http://127.0.0.1:' + PORT, opts)
+    .then(function() {
+      should.equal(typeof headers.authorization, 'string');
     });
   });
 
+  it('Test headers are sent correctly on GET request', function() {
+    var pouchDB = new PouchDB('http://127.0.0.1:' + PORT);
+    var opts = { ajax: { headers: { ick: "slick" } } };
+    return pouchDB.get("fake", opts).then(function() {
+      should.equal(headers.ick, "slick");
+    });
+  });
+
+  it('Test that we combine ajax options both globally and locally on GET',
+      function() {
+    var opts = { ajax: { headers: { aheader: "whyyes" } } };
+    var pouchDB = new PouchDB('http://127.0.0.1:' + PORT, opts);
+    var getOpts = { ajax: { headers: { ick: "slick", aheader: "override!" } } };
+    return pouchDB.get("fake", getOpts).
+    then(function() {
```

Comment:
Minor nit, we arent super consistent about these, but I think its far clearer having the then on the previous line

    return pouchDB.get('fake', getOpts).then(function() { 
      should.equal....
    });

Another minor nits are ' over ", and `var db = new PouchDB` instead of `pouchDB`